apps/dashboard/src/components/conditions-editor/variable-select.tsx

```diff
@@ -0,0 +1,265 @@
+          // Scroll down if option is below visible area
+          listElement.scrollTop = optionTop + optionHeight - containerHeight;
+        }
+      }
+    };
+
+    const next = () => {
+      if (hoveredOptionIndex === -1) {
+        setHoveredOptionIndex(0);
+        scrollToOption(0);
+      } else {
+        setHoveredOptionIndex((oldIndex) => {
+          const newIndex = oldIndex === options.length - 1 ? 0 : oldIndex + 1;
+          scrollToOption(newIndex);
+          return newIndex;
+        });
+      }
+    };
+
+    const prev = () => {
+      if (hoveredOptionIndex === -1) {
+        setHoveredOptionIndex(options.length - 1);
+        scrollToOption(options.length - 1);
+      } else {
+        setHoveredOptionIndex((oldIndex) => {
+          const newIndex = oldIndex === 0 ? options.length - 1 : oldIndex - 1;
+          scrollToOption(newIndex);
+          return newIndex;
+        });
+      }
+    };
+
+    return (
+      <div className="flex flex-col">
+        <input
+          className="sr-only"
+          ref={ref}
+          onFocusCapture={() => {
+            setHoveredOptionIndex(0);
+            scrollToOption(0);
+          }}
+          onKeyDown={(e) => {
+            if (e.key === 'ArrowDown') {
+              next();
+              e.preventDefault();
+            } else if (e.key === 'ArrowUp') {
+              prev();
+              e.preventDefault();
+            } else if (e.key === 'Enter') {
+              if (hoveredOptionIndex !== -1) {
+                onSelect(options[hoveredOptionIndex].value ?? '');
+                setHoveredOptionIndex(-1);
+              }
+            }
+          }}
+        />
+        <header className="flex items-center justify-between gap-1 border-b border-neutral-100 bg-neutral-50 p-1">
+          <span className="text-foreground-400 text-paragraph-2xs uppercase">Variables</span>
+          <KeyboardItem>{`{`}</KeyboardItem>
+        </header>
+        <ul
+          ref={listRef}
+          // relative is to set offset parent and is important to make the scroll and navigation work
+          className="relative flex max-h-[200px] flex-col gap-0.5 overflow-y-auto overflow-x-hidden p-1"
+        >
+          {options.map((option, index) => (
+            <li
+              className={cn(
+                'hover:bg-accent text-paragraph-xs flex cursor-pointer items-center gap-1 rounded-sm p-1',
+                hoveredOptionIndex === index ? 'bg-accent' : ''
+              )}
+              key={option.value}
+              value={option.value}
+              onClick={() => {
+                onSelect(option.value ?? '');
+              }}
+            >
+              <Code2 className="text-feature size-3 min-w-3" />
+              {option.label}
+              <CheckIcon
+                className={cn('ml-auto size-4', selectedValue === option.value ? 'opacity-50' : 'opacity-0')}
+              />
+            </li>
+          ))}
+        </ul>
```

Comment:
Our ControlInput autocomplete also has a search text highlight at the bottom which was quite helpful. (from code mirror) 
![CleanShot 2025-01-14 at 16 36 50@2x](https://github.com/user-attachments/assets/e0cc3a28-577a-491f-9482-b9c3695ccdfa)

Reply:
Is it complex to implement?

Reply:
it shouldn't be I'll take a look 👀